### DISCUSSION TITLE
Can't manage to push an image to dockerhub with ImageSpec

### DISCUSSION THREAD

**Discussion by ALBDN** [flyteorg/flyte#5420] [Q&A]
Hi!

I'm trying to use this great ImageSpec feature with docker hub registry but I'm facing this error:
`Failed with Unknown Exception <class 'Exception'> Reason: failed to run command envd build --path /tmp/flyte1ma3cexz/control_plane_metadata/local_flytekit/ab5988b3a69fba5645e9128922535299  --platform linux/amd64 --output type=image,name=alevelbdn/flytekit/flytekit:aLJkDnqKAzK_4X_sEcGCaA,push=true with error b'time="2024-05-24T10:32:55+02:00" level=fatal msg=exit app=envd error="failed to build the image: failed to build: failed to wait error group: failed to solve LLB: failed to solve: server message: insufficient_scope: authorization failed" version=v0.3.45\n'`

I know the error shows _insufficient_scope: authorization failed_ but I did a _docker login_ first which turned out to be successful. I've also tried to build the image and push it to the docker registry with:
`docker build -t alevelbdn/flytekit:v1 .`
`docker push alevelbdn/flytekit:v1`
and it worked without any problem.  I can find the image in my registry.

Now when trying with
`image_definition = ImageSpec(
   packages=["pandas", "numpy", "joblib", "scikit-learn", "xgboost"],
   registry="alevelbdn/flytekit",
   builder="envd"
)`
in my python script and pyflyte run with:
`pyflyte run --remote -p house-pricest -d development house_price_predictor.py house_price_predictor_trainer`
I end up with the above error.

My environment: Flyte is deployed on a local Kubernetes cluster with the flight-binary helm chart.
I can deploy a simple Flyte pipeline (without dependencies) and run it successfully on the cluster.

As I'm running out of idea It would be really nice if someone who faced the above issue or have any idea could advise...
Thank you!

**Comment by Future-Outlier** [comment]
Hi, @ALBDN  maybe you can try `docker login` first and then run the code again?

**Reply by ALBDN** [reply]
Hi Future-Outlier,
As explained in my post above I did a _docker login_  (as mentioned in the documentation) before running the code but it does not help.
